octoprint_mrbeam/util/material_csv_parser.py

```diff
@@ -219,12 +222,15 @@ def parse_csv(path=None, laserhead=MRBEAM):
                 pierce_time,
                 dithering,
             ]  # update current row values for next loop
-    converted_laserhead = model_id_to_csv_name(laserhead)
-    if converted_laserhead:
-        laserhead = converted_laserhead
-    if laserhead not in dictionary:
-        laserhead = DEFAULT_LASER
-    res = dict(materials=dictionary.get(laserhead, {}), laser_source=laserhead)
+    device_model_name = model_id_to_csv_name(device_model)
+    lh_to_device_mapping = ""
+    if device_model_name and device_model_name is MRB_DREAMCUT and laserhead_model == "S":
+        lh_to_device_mapping = MRB_DREAMCUT_S
+    elif device_model_name:
+        lh_to_device_mapping = device_model_name
+    if lh_to_device_mapping not in dictionary:
+        lh_to_device_mapping = DEFAULT_LASER
+    res = dict(materials=dictionary.get(lh_to_device_mapping, {}), laser_source=laserhead_model)
     return res
```

Comment:
I think this might be simplified by having tuple-keys in the `convert` dictionary:

```py
convert = {
  (mrbeam_model, laserhead) : spreadsheet_name
}
```

This way you don't need the extra mapping of laserhead to mrbeam_model which gets a bit confusing to read.

It is also more readable and shows which are the "permitted" combinations of device models and laserheads.
Aaand you can also have `("MRBEAM2_DC_S", DEFAULT_LASER)` as a possibility.

Reply:
Code has been refactored